tests/functional/object/rgw/test_rgw_sts.py

```
@@ -524,3 +521,346 @@ def test_rgw_sts_configuration(
+            ), f"Unexpected error code for invalid session token: {error_code}"
+
+        # Step 10: Wait for STS token to expire
+        logger.info(f"Step 10: Waiting {token_duration + 10}s for STS token to expire")
+        time.sleep(token_duration + 10)
+        logger.info("Step 10: Token expiration wait complete")
+
+        # Step 11: Get bucket content after token expiration - expect failure
+        # Use TimeoutSampler to retry for up to 120s to handle infra timing jitter
+        logger.info("Step 11: Listing bucket content after token expiration")
+
+        def _is_token_expired():
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_is_token_expired`

Add return type annotation: `bool | None`

(ANN202)